### PR TITLE
Replace CRA with Vite in React generator

### PR DESCRIPTION
**Pull request by jvasseur**
This is the same as #1374, but implemented as a complete replacement of CRA in the React generator.

### CODE REVIEW

File: generators/react/templates/base/README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ In the project directory, you can run:
 ### `yarn start`
 
 Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+Open [http://localhost:xxxx](http://localhost:xxxx) to view it in the browser.
```
Comment (Mysh3ll) [inline generators/react/templates/base/README.md:12]: Peux être mettre 3000, vu que dans `generators/react/templates/base/vite.config.ts.ejs` il y a `port: 3000`.

File: generators/react/templates/base/package.json.ejs
Comment (Mysh3ll) [inline generators/react/templates/base/package.json.ejs:1]: L'indentation ?!

File: generators/react/templates/base/src/index.css
Comment (Mysh3ll) [inline generators/react/templates/base/src/index.css:1]: L'indentation ?!

**Review by alexandre-charon — APPROVED** [review]

File: generators/react/templates/base/src/vite-env.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+/// <reference types="vite/client" />
```
Comment (blatinier) [inline generators/react/templates/base/src/vite-env.d.ts:1]: Un fichier avec juste une ligne commenté sans explication c'est un peu aride :(

**Review by blatinier — APPROVED** [review]

File: generators/react/templates/base/src/vite-env.d.ts (hunk 2 above)
Comment (jvasseur) [reply generators/react/templates/base/src/vite-env.d.ts:1]: It's a typescript [directive](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html) that is generated like this by `create-vite`.

It's a technicality of how typescript work and this file should never be touched, not sure if it's worth adding a comment to explain it.

**Pull request merged**